### ISSUE TITLE
Embroider 1.9.0 breaks all `this` context in component templates

### ISSUE THREAD

**Issue by rwwagner90** [embroider-build/embroider#1287]
If I update to Embroider 1.9.0 and try to use `this.foo` in my template, it tells me it is undefined. Any ideas what is going on here?

**Comment by rwwagner90** [comment]
Seems to be related to ember-cli-typescript

**Comment by lifeart** [comment]
Looks related to https://github.com/embroider-build/embroider/issues/1261

**Comment by NullVoxPopuli** [comment]
As a counter example (meaning, actual problem is something very specific, perhaps?) I have embroider 1.9.0 + TS working here: https://github.com/NullVoxPopuli/limber/blob/main/frontend/ember-cli-build.js
_however_ every single one of my components is using `<template>`

**Comment by rwwagner90** [comment]
We think this might be related to ember-svg-jar as well. Something with running it and TS.

**Comment by rwwagner90** [comment]
Not related to svg-jar. It's just using TS for a component and trying to use Embroider 1.9.0 makes things go 💥 . Reproduction here https://github.com/rwwagner90/embroider-svg-ts-bug

**Comment by ef4** [comment]
This is a duplicate of #1273 which is fixed on `main`. Working on getting it released.

**Comment by rwwagner90** [comment]
@ef4 any idea when a release will be cut?

**Comment by ef4** [comment]
I'm actively working on it, but I'm doing it carefully by hand because we have a new unproven release process that lets us do semver major releases in only the packages that need them.

This bug fix is in `@embroider/compat`, the next release of that package is planned to be 2.0 because of changes to this supported ember versions, but we can circle back around and back port the bug fix to 1.9 if that becomes important.

The breaking change is that we dropped support for unsupported ember releases. So the oldest supported ember will become 3.28.

**Comment by rwwagner90** [comment]
@ef4 I think a major breaking release sounds fine to me. Ember 3 is getting older now, and I think we'll want people to be on the latest and greatest to get all the nice things.

**Comment by rwwagner90** [comment]
> I hoped to get it done today but I’m not there yet.
> 
> (In general, when blocked on an upstream bug I think patch-package is a good solution. Also pnpm has built-in support for the same kind of thing.)

We have decided to just not use Embroider for now, so I am unblocked.
